### PR TITLE
fix(route): throw on attempt to fulfill with redirect in WebKit

### PR DESCRIPTION
**Pull request by yury-s**
#1489

### CODE REVIEW

File: src/server/network.ts [hunk 1]
```diff
@@ -212,6 +212,8 @@ export class Route {
   }
 
   async fulfill(response: { status?: number, headers?: types.HeadersArray, body?: string, isBase64?: boolean }) {
+    if (response.status && 300 <= response.status && response.status < 400)
+      throw new Error('Cannot fulfill with redirect status');
```
Comment (dgozman) [inline src/server/network.ts:216]: nit: let's include the bad status in the error message

**Review by dgozman — APPROVED** [review]

File: src/server/network.ts (hunk 1 above)
Comment (yury-s) [reply src/server/network.ts:216]: Done.

**Comment by JoelEinbinder** [comment]
Redirects are functionally different to fulfilling the redirected content. Consider intercepting a module script. If we fetch the script ourselves instead of providing a redirect, all of the relative imports in that script will point to the wrong place. Can we leave the current behavior for browsers in which it works?

**Comment by yury-s** [comment]
@JoelEinbinder In the example with a module script intercepting all requests with a given prefix should solve the issue or am I missing something? I certainly can leave existing behavior for FF and CR (not sure to what extent it works though) but it'd be nice if the method behaved consistently across browsers.

**Comment by JoelEinbinder** [comment]
if I have `www.example.com/realdata.jsm`, and I want to redirect to replace just this file with a new module located at at `www.example.com/for-testing/mockdata.jsm`. `mockdata.jsm` imports `./foo.jsm` and `./bar.jsm`. If I have a redirect, they will access `for-testing/foo.jsm` and `for-testing/bar.jsm`. If I fulfill the content of `mockdata.jsm`, they will attempt to access the toplevel `/foo.jsm` and `bar.jsm`. I can't use a prefix to know which files were from `mockdata.jsm` and should have different paths.

That's just one example. It also effects `import.meta`, relative urls in css, and I imagine a lot of other subtle scenarios.

**Comment by dgozman** [comment]
> if I have `www.example.com/realdata.jsm`, and I want to redirect to replace just this file with a new module located at at `www.example.com/for-testing/mockdata.jsm`. `mockdata.jsm` imports `./foo.jsm` and `./bar.jsm`. If I have a redirect, they will access `for-testing/foo.jsm` and `for-testing/bar.jsm`. If I fulfill the content of `mockdata.jsm`, they will attempt to access the toplevel `/foo.jsm` and `bar.jsm`. I can't use a prefix to know which files were from `mockdata.jsm` and should have different paths.
> 
> That's just one example. It also effects `import.meta`, relative urls in css, and I imagine a lot of other subtle scenarios.

Sounds like a pretty convoluted usecase. You basically end up with two different `foo.jsm`, one from the real code, and one from the mock data. Could be reasonable, but perhaps we could leave this to the user to sort out?

I think we should strive for cross-browser compatibility as much as we can.

**Comment by JoelEinbinder** [comment]
I've done the module redirecting case. The linked issue has multiple users that want url redirecting, one of which is currently successfully using a 301 redirect. This patch is a regression and will break people.

**Comment by yury-s** [comment]
Ok, let's limit this to webkit.

**Comment by yury-s** [comment]
Updated the code to throw only in webkit if the client tries to fulfill with a redirect. Updated tests to reflect current behavior across browsers.

**Review by dgozman — APPROVED** [review]

**Pull request merged**